File: apps/android/app/src/main/java/com/propertyvault/app/data/units/Unit.kt
package com.propertyvault.app.data.units

// Domain model -- mirrors packages/types/src/portfolio.ts's Unit interface. Named PropertyUnit
// (not Unit) to avoid colliding with Kotlin's own `Unit` type across this file's own signatures.
data class PropertyUnit(
    val id: String,
    val propertyId: String,
    val orgId: String,
    val unitLabel: String,
    val bedrooms: Int?,
    val bathrooms: Int?,
    val sizeSqm: Double?,
    val marketRent: Double?,
    val status: String,
)

File: apps/ios/Sources/Proplyst/Domain/Unit.swift
import Foundation

/// Mirrors `apps/android`'s `PropertyUnit` domain model (named to avoid colliding with Swift's
/// own `Unit` type, the same reason Android's own model avoids the bare name `Unit`).
struct PropertyUnit: Identifiable, Codable, Equatable, Sendable {
    let id: String
    let propertyId: String
    let orgId: String
    let unitLabel: String
    let status: String
    let sizeSqm: Double?
    let marketRent: Double?

    enum CodingKeys: String, CodingKey {
        case id
        case propertyId = "property_id"
        case orgId = "org_id"
        case unitLabel = "unit_label"
        case status
        case sizeSqm = "size_sqm"
        case marketRent = "market_rent"
    }
}
